### PR TITLE
Fix broken links in documentation.

### PR DESCRIPTION
**Pull request by patrickdavey**
This fixes a few broken links. One of which (the overview-of-exercism
link) seems like it should now point to the overview-architecture page,
which is what I've made it do.

Thanks for the great website! Love Exercism!

### CODE REVIEW

File: about/architecture.md [hunk 1]
```diff
@@ -9,7 +9,7 @@ The main components that make up Exercism are:
 
 ### The Website
 
-The [exercism.io](http://exercism.io/) website is a vital part of the [product](https://github.com/exercism/exercism.io/blob/master/docs/overview-of-exercism.md#the-product). The community uses the site to:
+The [exercism.io](http://exercism.io/) website is a vital part of the [product](https://github.com/exercism/exercism.io/blob/master/docs/overview-architecture.md). The community uses the site to:
```
Comment (kytrinyx) [inline about/architecture.md]: This should probably be https://github.com/exercism/docs/blob/master/about/README.md#the-product

**Review by kytrinyx — CHANGES_REQUESTED** [review]
Thanks for catching these! I've suggested an alternate link for one of them, other than that this looks great.

**Comment by patrickdavey** [comment]
Thanks @kytrinyx .  I fixed the link and pushed it back up (altering the commit message to remove the part about the link I wasn't sure about). I did force push it up.. I thought an extra commit would just be extra noise.

**Review by kytrinyx — APPROVED** [review]

**Comment by kytrinyx** [comment]
Sweet, thanks for the quick turnaround on this! 🌷

**Pull request merged**